### PR TITLE
Support resolution change

### PR DESCRIPTION
**Pull request by Dinduks**
I thought about "performances" issues and made this little script:

```python
#!/usr/bin/env python

import windows
import time

print(time.time())

for i in range(0, 100):
    windows.get_screen_resolution()[0]
    windows.get_screen_resolution()[1]

print(time.time())
```

Running prints:
```
1430775714.23
1430775714.66
```

So I guess it's fine calling `windows.get_screen_resolution()` on every screen resize.

### CODE REVIEW

**Comment by joaquincasares** [comment]
I'd actually want for that to be configurable for unique setups where it's harder for us to detect screen layouts, but do like the intent. Could you make it so that someone can override the value using a configuration parameter? Thanks!

**Comment by Dinduks** [comment]
Which value(s) do you exactly want configurable, and how would you like to configure them?

**Comment by joaquincasares** [comment]
usable_screen_width and usable_screen_height. I'd probably go with creating a helper function that checks to see if the values are `None` in the config. If the are not set, use `windows.get_screen_resolution()`. If they are set, use what's in the config.

**Comment by Dinduks** [comment]
I can't think of a case where a user would like to set the resolution themselves, while the system can give it to them (for free :smile:).  
Do you have any in mind?

**Comment by joaquincasares** [comment]
Heh, fair point! :)

I was thinking of dual-screen support. Right now I've stopped using ration as much since it was acting wonky when I was running two screens. (I forget the exact issues.) I figured setting the resolution manually would be foundation of easy access into dual-screen support, but never began testing it.

Thoughts?

**Comment by Dinduks** [comment]
I don't dual screen at the moment and I don't know the code base well enough to comment.

I suggest creating an issue for that. I'll (try to) take a look at it when I have some free time. :)

**Comment by joaquincasares** [comment]
No worries on not having a dual-monitor setup. Just refactor your code to use a helper method that will use the CONFIG resolution if provided, if not, then dynamically use the current screen resolution as you had intended. 

That way we can add your use current case into the master while leaving the pathway open for future dual-monitor testing.

**Comment by Dinduks** [comment]
Wouldn't it be more reasonable to keep the code the way it is, then update it to use the config after the config is implemented?

**Comment by joaquincasares** [comment]
Sorry, I'm not sure I follow. Could you rephrase that?

**Comment by Dinduks** [comment]
I believe that it's better to merge my code right now, then, later, implement the custom resolution feature and update the code to use it.

**Comment by joaquincasares** [comment]
Sure I'd be down with that except it goes against one of my most recent commits to make more, rather than less, variables configurable: 7c4a8e65ad45d79208301dfac363b54ee22d99db.

It took me a while of digging around the code to follow where special variables I was looking at tweaking were being setup. This commit, along with all the recent ones, pushed all variables to the top for quicker new user development. Having all variables in one place would also continue to passively drive contributors like you to a seemingly stale project with minimal ramp-up friction.

Merging the code as is would go against this direction while including a helper function that could accept manual overrides would continue along the path while advocating higher readability.

**Comment by Dinduks** [comment]
I understand what you're trying to do, and while I believe the paddings and margins settings would be desirable for some users, I believe the usefulness of setting resolution manually is close to none and just adds bloat (e.g. helper function), because the OS can give the exact resolution of the screen(s) with no possibility of human mistake.

Regarding the dual screen issue, as we're not sure it'll be fixed by setting the resolution manually, I believe it should be investigated further?

**Comment by joaquincasares** [comment]
I'm sorry, I'm going to stand by my request to make a helper function here to keep the main variables clearly marked and easily configurable, even if that means adding a helper function. If you update your pull request, I'll gladly accept it.

Indeed, dual screen support has to be investigated, but that topic is best kept to another ticket.

**Comment by Dinduks** [comment]
And I stand by my opinion that the screen resolution should be provided by the OS instead of being specified by the user (as it is not a *personal preference* such as margins or the size of the grid).

Feel free to use this PR's code as a base for other features you believe relevant.

Thanks.
